src/CovidDataDashboard/CovidDataClip.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { FunctionComponent } from "react"
+import { ActivityIndicator, StyleSheet, View } from "react-native"
+import { useTranslation } from "react-i18next"
+
+import CovidDataStateTrend from "./CovidDataStateTrend"
+import { Text } from "../components"
+import { useConfigurationContext } from "../ConfigurationContext"
+import {
+  CovidDataRequestStatus,
+  useCovidDataContext,
+} from "../CovidDataContext"
+
+import {
+  Affordances,
+  Colors,
+  Iconography,
+  Spacing,
+  Typography,
+} from "../styles"
+
+const LoadingIndicator = () => {
+  return (
+    <View style={style.activityIndicatorContainer}>
+      <ActivityIndicator
+        size={"large"}
+        color={Colors.neutral100}
+        style={style.activityIndicator}
+        testID={"loading-indicator"}
+      />
+    </View>
+  )
+}
+
+const CovidDataClip: FunctionComponent = () => {
+  const { stateAbbreviation } = useConfigurationContext()
+  const { t } = useTranslation()
+  const {
+    covidDataRequest: { data, status },
+  } = useCovidDataContext()
+
+  if (stateAbbreviation === null) {
```

Comment:
I wouldnt change anything in this pr, but this stateAbbreviation null checking jumps out as a bit of a smell. I wonder if there is a way to pull this up the chain and have some sort of `isAbleToFetchCovidData` validation and only pass data to the presentational components?  Id be afraid of making things too complicated, but seems like one of the harder parts of this feature in general, so might make sense to think through it on a follow up.

Reply:
Right there with you, I'm moving most of it to the data context and adding a few tests there now will come in the next PR 😄